modules/http-api/src/main/java/org/mule/runtime/http/api/server/HttpRequestParsingException.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.http.api.server;
+
+/**
+ * Exception thrown when there is a problem processing an incoming HttpRequest to generate a MuleEvent.
+ *
+ * @since 1.2
+ */
+public class HttpRequestParsingException extends Exception {
```

Comment:
If this is not used then the hierarchy is too much. We can have just `DecodingException` for now and create the hierarchy later if it makes sense.